pkg/lib/notifications.ts

```diff
@@ -115,28 +115,34 @@ Usage:
 
 */
 
-import cockpit from "cockpit";
+import cockpit, { JsonValue, JsonObject } from "cockpit";
 import { dequal } from 'dequal/lite';
 
-class PageStatus {
+export interface Status {
+    type?: string | null;
+    title?: string;
+    details?: JsonObject;
+}
+
+class PageStatus extends EventTarget {
```

Comment:
That's sus -- that should be a `cockpit.EventSource`, not the standard JS platform one, no? (I mean, I'd **like** it to be the standard interface, but alas, we have a third one now -- pkg/lib/cockpit/event.ts which is `EventEmitter`.

Reply:
The call to `cockpit.event_target` is gone, this is really a proper standard EventTarget now.

cockpit.EventSource is nicer because of the EventMap, but we can't derive from it since it is not a class. We can make that possible by turning it into a class, but I think there is no real motivation for that when EventTarget already exists.

Reply:
Nice!

Reply:
(it's the first time that we do that)

Reply:
I almost did that for ShellState et al, but our own EventEmitter is sufficiently nicer, and there was no compat to keep.